### PR TITLE
Adds logo link and functionality 

### LINKED ISSUE #394: Logo link and other functionality
When a user clicks the "Reddit Image Viewer" text in the header, it should clear the search, which will load the default sub `itookapicture`.

- [ ] Create clear function
- [ ] Hook up to header "logo" text

### PR DESCRIPTION
**Pull request by AloyBranCheung**
Closes #394

### Description

What does your feature do? Give some context...
This features allows the standard UX behavior of clicking the brand logo to head back to start/homepage. In the case of this app, it clears the search bar and returns  `itookapicture` as the default start images

Implementation: 
I moved the input state to the parent component so that the `<Header />` component could also access the state. That way, when the brand icon is clicked, the search bar can be set to a blank. 

To show `itookapicture` images when clicked, I used a template literal to check if the state is empty to default to `itookapicture`

### Screenshot

If possible, add some screenshots of your feature.

### Verification

How will a stakeholder test this?

1.Load the homepage
1.Search for subreddit pictures, try AnimalCrossing (i love that game)
1.Click the brand logo, should navigate you back to `itookapicture` subreddit

### CODE REVIEW

**Review by gregrickaby — CHANGES_REQUESTED** [review]
@AloyBranCheung Thank you for doing this. I have two things...

Rather than lifting state up to the `<Header />`, can you move the state management to the `<RedditProvider />` instead? I'd rather not have several locations handling state if possible. 

Also, when you click the logo, the feed doesn't reset to `itookapicture`...it just clears the `<Search />` component.

**Comment by gregrickaby** [comment]
@AloyBranCheung I anticipated this, so I just pushed up ENV var instructions 😄 

https://github.com/gregrickaby/reddit-image-viewer/blob/main/CONTRIBUTING.md#env-variables

**Comment by AloyBranCheung** [comment]
I see it now thanks for instructions. Sorry for trying to get away with just network traffic 😂😂😭😭😭

**Comment by gregrickaby** [comment]
@AloyBranCheung FYI: #400 changed which ENV vars are required. Please view the updated [CONTRIBUTING](https://github.com/gregrickaby/reddit-image-viewer/blob/main/CONTRIBUTING.md#env-variables) doc.

Sorry in advanced for all the ENV switcharoos, but the `Bearer` token kept expiring, which broke the search ☹️ It should be stable now!

**Comment by AloyBranCheung** [comment]
Gotchu thanks for letting me know. I just synced the repo and updated my branch. Hopefully I have time tonight to start cracking at it.

**Comment by AloyBranCheung** [comment]
@gregrickaby I think this should work

File: components/RedditProvider.tsx [hunk 1]
```diff
@@ -7,6 +7,8 @@ export interface RedditProviderProps {
   subReddit: any
   setSort: (sort: string) => void
   setSubreddit: (subReddit: {}) => void
+  searchInput: string
+  setSearchInput: React.Dispatch<React.SetStateAction<string>>
```
Comment (gregrickaby) [inline components/RedditProvider.tsx:11]: @AloyBranCheung While I appreciate the verboseness of this type, I'd rather it stay in line with the rest of the project. Can you update it to something like: `(searchInput: string) => void` 

Thank you

File: components/Search.tsx [hunk 2]
```diff
@@ -18,28 +17,32 @@ const useStyles = createStyles(() => ({
...
-  const [debounced] = useDebouncedValue(value, 300)
+
+  const [debounced] = useDebouncedValue(searchInput, 300)
   const {data: results} = useSWR(`/api/search?term=${debounced}`, fetcher, {
     revalidateIfStale: false,
     revalidateOnFocus: false,
     revalidateOnMount: false
   })
 
+  const handleChange = (string: string) => {
```
Comment (gregrickaby) [inline components/Search.tsx:30]: @AloyBranCheung Can you please make this named function?

```ts
function handleChange(string: string) {
  setSearchInput(string)
}

**Review by gregrickaby — CHANGES_REQUESTED** [review]

**Review by gregrickaby — APPROVED** [review]
@AloyBranCheung 👏🏻👏🏻 Way to stick with it. Thank you so much for your contribution! 🥳🙏🏻

**Comment by AloyBranCheung** [comment]
@gregrickaby thanks for your patience and guidance :)

**Pull request merged**